### DISCUSSION TITLE
Execute code as the server (admin)

### DISCUSSION THREAD

**Discussion by AshtonHannon** [pocketbase/pocketbase#4337] [Q&A] [answered]
I'm sure this has been answered elsewhere, but I can't for the life of me find an answer to what I'm looking for...

I'm using SvelteKit + Pocketbase, and I have a registration form. Upon submitting of the registration form, I want to check if a user has already been created with the submitted email. How can I (using the JS-SDK) check the database as an admin/server, skipping the API rules that I want in place for all other instances, in order to check if a user with the given email has been created?

Any help is appreciated.

**Comment by ganigeorgiev** [comment]
>  I want to check if a user has already been created with the submitted email.

You don't have to do it manually because there is already a builtin unique constraint for the email field. If someone tries to register with already registered email they should get an error response.

For using the SDK as admin you can check the example admin client from https://github.com/pocketbase/pocketbase/discussions/3792#discussioncomment-7621130.
If you want to fetch a record by a non-id field you can use the `getFirstListItem` helper:
```js
await pb.collection("example").getFirstListItem("someField='example'")
```

**Reply by AshtonHannon** [reply]
Well the reason I am doing it manually is for frontend feedback on the form (i.e. "User already exists with this email")... Are you suggesting the better way to do it would be to catch the error when trying to create the new user, and then show the frontend feedback from that?

Correct me if I'm wrong, but from the thread you sent me, I can just create a separate "admin authenticated" client, and use it as "adminPB" throughout the app?

**Reply by ganigeorgiev** [reply, marked as answer]
> Are you suggesting the better way to do it would be to catch the error when trying to create the new user, and then show the frontend feedback from that?

Yes, you could inspect the `err.response` object (for more details about the available error fields please refer to [Error handling](https://github.com/pocketbase/js-sdk?tab=readme-ov-file#error-handling)).

> Correct me if I'm wrong, but from the thread you sent me, I can just create a separate "admin authenticated" client, and use it as "adminPB" throughout the app?

Yes.

**Reply by AshtonHannon** [reply]
Thanks so much.
